crates/lakekeeper/src/service/tasks/mod.rs

```diff
@@ -208,10 +273,11 @@ pub struct TaskInput {
 /// Metadata stored for each task in the database backend.
 /// This is separate from the task payload, which is specific to each task type.
 pub struct TaskMetadata {
-    pub warehouse_id: WarehouseId,
+    pub warehouse_id: Option<WarehouseId>,
+    pub project_id: ProjectId,
     pub parent_task_id: Option<TaskId>,
     pub entity_id: EntityId,
-    pub entity_name: Vec<String>,
+    pub entity_name: Option<Vec<String>>,
```

Comment:
This doesn't feel right. If entity_id is known, then also an entity_name should be known. In Lakekeeper all entities are named, so we should know either both or none.